state-chain/amm-math/src/lib.rs

```diff
@@ -108,6 +109,27 @@ pub fn output_amount_ceil(input: Amount, price: Price) -> Amount {
 	mul_div_ceil(input, price, U256::one() << PRICE_FRACTIONAL_BITS)
 }
 
+pub fn price_from_usd_fine_amount(price_usd: AssetAmount) -> Price {
+	Price::from(price_usd) << PRICE_FRACTIONAL_BITS
+}
+
+/// Get the price of an asset given its dollar amount and decimals.
+/// Using u32 for dollar_amount so it is type safe against fine amounts (u128).
+pub fn price_from_usd(dollar_amount: u32, decimals: u32) -> Price {
+	if decimals == 0 || dollar_amount == 0 {
+		return U256::zero();
+	}
+
+	(U256::from(dollar_amount).saturating_mul(U256::from(10u128.pow(USD_DECIMALS))) <<
+		PRICE_FRACTIONAL_BITS) /
+		10u128.pow(decimals)
```

Comment:
If we handle the assets inside here, then we can collapse this, because we always use more than 6 decimals.

The calculation will be just `(Price::from(dollar_amount) << PRICE_FRACTIONAL_BITS) / 10.pow(decimals - USD_DECIMALS)`